akka-projection-samples/src/main/scala/akka/projection/scaladsl/jdbc/JdbcOffsetStore.scala

```diff
@@ -0,0 +1,36 @@
+package akka.projection.scaladsl.jdbc
+
+import akka.persistence.query.{ Offset, TimeBasedUUID }
+
+import scala.concurrent.Future
+
+trait JdbcOffsetStore[O] {
+  def projectionId: String
+  def readOffset(connection: Connection): Future[Option[O]]
+  def saveOffset(connection: Connection, offset: O): Unit
+}
```

Comment:
This is a very elementary impl just to exercise some ideas. 

I was imagining if it would make sense to always save the offset as a String and have a column for the manifest. 

So for instance, a `JdbcOffsetStoreLong` impl will know how to read the string and convert it to Long using the manifest. 

The implementation tailored for Akka Persistence Offset will know how to convert it to `Sequence` or to `TimeBasedUUID`. 

For the user, it will have the added value that they will be able to go to one table and see all the offsets in a readable format (String). I'm mainly thinking on the case in which you have Akka Persistence Journal and Kafka offsets in the same application. 

I would also prefer to not have different table definitions to manage. One table to rule them all.